src/auth/AuthContextProvider.jsx

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable react/react-in-jsx-scope */
+/* eslint-disable react/jsx-no-constructed-context-values */
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { AuthContext } from './AuthContext';
+import checkIfUserLogged from '../helpers/checkIfUserLogged';
+
+export default function AuthContextProvider(props) {
+  const [logged, setLogged] = useState(checkIfUserLogged());
```

Comment:
It's a bad practice to put heavy operations here (localstorage reading) as it will call localstorage every time component rerenders. Check out state lazy loading 

https://kentcdodds.com/blog/use-state-lazy-initialization-and-function-updates

Reply:
I have deleted this file, and after all changes I think this problem is gone. I am reading from localStorage only in `useState` in `App.jsx`